### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by nomo-nomad**
add .product(name:, package:) to dependency declarations

### CODE REVIEW

**Review by MaxDesiatov — APPROVED** [review]
Seems legit 👍

File: README.md [hunk 1]
```diff
@@ -24,9 +24,12 @@ dependencies: [
...
 targets: [
-    .target(name: "MyAwesomePackage", dependencies: ["OpenCombine",
-                                                     "OpenCombineDispatch",
-                                                     "OpenCombineFoundation"])
+    .target(name: "MyAwesomePackage",
+    	    dependencies: ["OpenCombine",
+                           .product(name: "OpenCombineFoundation", 
+			   	    package: "OpenCombine"),
+                           .product(name: "OpenCombineDispatch",
+			   	    package: "OpenCombine")]),
```
Comment (broadwaylamb) [inline README.md]: Please align the lines.

**Review by broadwaylamb — COMMENTED** [review]
Could you please explain the reason behind this change?

**Comment by MaxDesiatov** [comment]
These are usually required when using Swift 5.2 or later to enable [target based dependency resolution](https://github.com/apple/swift-evolution/blob/main/proposals/0226-package-manager-target-based-dep-resolution.md).

**Comment by broadwaylamb** [comment]
I see. In that case, LGTM, but the lines need to be aligned.

File: README.md (hunk 1 above)
Comment (nomo-nomad) [reply README.md]: Sorry, I'm not sure exactly how you'd format the lines, is this better?
```
targets: [
    .target(name: "MyAwesomePackage",
    	    dependencies: ["OpenCombine",
                           .product(name: "OpenCombineFoundation", package: "OpenCombine"),
                           .product(name: "OpenCombineDispatch", package: "OpenCombine")]),
]
```

File: README.md [hunk 2]
```diff
@@ -24,9 +24,10 @@ dependencies: [
     .package(url: "https://github.com/OpenCombine/OpenCombine.git", from: "0.11.0")
 ],
 targets: [
-    .target(name: "MyAwesomePackage", dependencies: ["OpenCombine",
-                                                     "OpenCombineDispatch",
-                                                     "OpenCombineFoundation"])
+    .target(name: "MyAwesomePackage",
+    	    dependencies: ["OpenCombine",
+                           .product(name: "OpenCombineFoundation", package: "OpenCombine"),
+                           .product(name: "OpenCombineDispatch", package: "OpenCombine")]),
```
Comment (MaxDesiatov) [inline README.md]:
```suggestion
    .target(
        name: "MyAwesomePackage",
        dependencies: [
            "OpenCombine",
            .product(name: "OpenCombineFoundation", package: "OpenCombine"),
            .product(name: "OpenCombineDispatch", package: "OpenCombine")
        ]
    ),
```

**Review by broadwaylamb — APPROVED** [review]
Thanks!

**Pull request merged**